apps/meteor/app/livechat/server/lib/guests.ts

```diff
@@ -160,3 +165,92 @@ async function cleanGuestHistory(token: string) {
 	await LivechatInquiry.removeByIds(livechatInquiries.map(({ _id }) => _id));
 	void notifyOnLivechatInquiryChanged(livechatInquiries, 'removed');
 }
+
+export async function getLivechatRoomGuestInfo(room: IOmnichannelRoom) {
+	const visitor = await LivechatVisitors.findOneEnabledById(room.v._id);
+	if (!visitor) {
+		throw new Error('error-invalid-visitor');
+	}
+
+	const agent = room.servedBy?._id ? await Users.findOneById(room.servedBy?._id) : null;
+
+	const ua = new UAParser();
+	ua.setUA(visitor.userAgent || '');
+
+	const postData: ICRMData = {
+		_id: room._id,
+		label: room.fname || room.label, // using same field for compatibility
+		topic: room.topic,
+		createdAt: room.ts,
+		lastMessageAt: room.lm,
+		tags: room.tags,
+		customFields: room.livechatData,
+		visitor: {
+			_id: visitor._id,
+			token: visitor.token,
+			name: visitor.name,
+			username: visitor.username,
+			department: visitor.department,
+			ip: visitor.ip,
+			os: ua.getOS().name && `${ua.getOS().name} ${ua.getOS().version}`,
+			browser: ua.getBrowser().name && `${ua.getBrowser().name} ${ua.getBrowser().version}`,
+			customFields: visitor.livechatData,
+		},
+	};
+
+	if (agent) {
+		const customFields = parseAgentCustomFields(agent.customFields);
+
+		postData.agent = {
+			_id: agent._id,
+			username: agent.username,
+			name: agent.name,
+			...(customFields && { customFields }),
+		};
+
+		if (agent.emails && agent.emails.length > 0) {
+			postData.agent.email = agent.emails[0].address;
+		}
+	}
+
+	if (room.crmData) {
+		postData.crmData = room.crmData;
+	}
+
+	if (visitor.visitorEmails && visitor.visitorEmails.length > 0) {
+		postData.visitor.email = visitor.visitorEmails;
+	}
+	if (visitor.phone && visitor.phone.length > 0) {
+		postData.visitor.phone = visitor.phone;
+	}
+
+	return postData;
+}
+
+function notifyRoomVisitorChange(roomId: string, visitor: ILivechatVisitor) {
+	void api.broadcast('omnichannel.room', roomId, {
+		type: 'visitorData',
+		visitor,
+	});
+}
+
+// TODO: Remove this function in next major
```

Comment:
just do it now.. internal functions don't need to wait for major releases.

Reply:
Oh wasn't "major", was "next task" 🤦🏽 

anyways, removed